django/db/migrations/serializer.py

```diff
@@ -198,7 +198,12 @@ def serialize(self):
 class ModelFieldSerializer(DeconstructableSerializer):
     def serialize(self):
         attr_name, path, args, kwargs = self.value.deconstruct()
-        return self.serialize_deconstructed(path, args, kwargs)
+        try:
+            return self.serialize_deconstructed(path, args, kwargs)
+        except ValueError as e:
+            msg = 'During serialization of the field'
+            e.args = ('{} {} the following blew up: {}'.format(msg, self.value, e.args[0]), ) + e.args[1:]
+            raise e
```

Comment:
I don't see a reason for modifying a source exception, you can use:
```python
raise ValueError(...) from e
```
Also `blew up` is not a appropriate wording and a new exception is not more informative because it refers to the field class `... the field <django.db.models.fields.CharField>` instead of `<app_label>.<model_name>.<field_name>`, maybe:
```python
raise ValueError('Error during %s serializing: %s' % (field, e)) from e
```
I don't have a quick answer how to get a field path.

Reply:
@felixxm thanks for the feedback. I have changed the error message as you suggested. But I have kept the `.format` notation instead of `%s`, if that should also change, please let me know. I will look into how to get the field path before I update with a new commit.

Reply:
@felixxm  `self.value` shows the field path in the <app label>.<model name>.<field name> form. I have tested this by starting a django project using the amended source code and attempting to makemigrations for an app with a model that has a field that uses a lambda expression. 

I have had trouble getting the test to verify this behavior though, since when I try to import a model from `migrations.models` and serialize it, the field is treated as a `DeferredAttribute` instead of a model field and so doesn't get handled by the `ModelFieldSerializer`. When I tested in a new project I was able to get the desired behavior by changing a field on a model that already had migrations, so the only thing that had to be serialized was the field which I amended to use a lambda expression as its default. At this stage, I don't see how to achieve that in a test. Any suggestions?

Reply:
By making a custom field the test error message shows the desired field path.

Reply:
No it's not, without this patch it raises, e.g.
```
ValueError: Cannot serialize function: lambda
```
with this patch:
```
ValueError: Error during <django.db.models.fields.IntegerField> serializing: Cannot serialize function: lambda
```
It's not more descriptive, IMO. You can have hundreds of `IntegerField`s in dozens of apps. I would expect:
```
ValueError: Error during 'test_app.models.MyModel.field_1.default' serializing: Cannot serialize function: lambda
```
I don't see much value in this change if it's not feasible to get `<app label>.<model name>.<field name>.<parameter>` or at least `<app label>.<model name>.<field name>`.

Maybe we should fix this in `FunctionTypeSerializer`, it should be doable to get at least `test_app.models.MyModel` from `__qualname__` and `__module__` :detective:

Reply:
@felixxm Yes, I see your point. However, when the field is attached to a model the `__str__` method in the `Field` subclass will present the desired message.

I am trying to figure out how to test it properly so that the desired message shows up in the test. It will require a migrated model which then has the `UnserializableField` attached to it. Or alternatively altering an existing field on a migrated model to use a lambda. Otherwise the model serialization will fail instead of the field serialization. Even though, in that case, it would be helpful to say why the model is failing if it is because of a lambda expression in one of its fields.

I will also look into your suggestion about the `FunctionTypeSerializer`- thank you :+1:

Reply:
The `FunctionTypeSerializer` has the same issue, if the field isn't part of a model.

I need to figure out how to add the `UnserializableField` to a migrated model, within a test. So that the field is a member of a model, and that the model appears as though it is already migrated, and that the only change is to add the `UnserializableField`. Any ideas on how to do this?

Reply:
@felixxm the latest commit seems to have gotten the desired result.

Reply:
I'm sorry but it's still the same :shrug: . You can manipulate with tests to create an unreachable state where you will get an expected message but that's not the correct solution.

Please check an [attached project ](https://code.djangoproject.com/raw-attachment/ticket/25370/ticket_25370.tar.gz) and try to run:
```
$> python manage.py makemigrations
```
it raises:
```
ValueError: Error during <django.db.models.fields.IntegerField> serializing: Cannot serialize function: lambda
```

Reply:
Hi @felixxm, you were right, the error message can be constructed in the `FunctionTypeSerializer` - but I had trouble getting the field name, any ideas on that one? BTW thanks for the welcome aboard message :bowtie:

Reply:
Thanks for updates :+1: 

> ... but I had trouble getting the field name, any ideas on that one?

Unfortunately not, moreover I'm afraid that we will not be able to get `<app label>.<model name>.<field name>` or even `<app label>.<model name>` in a reliable way. We serialize `field` from `django.db.models` not a model attribute, that's why it's complicated or even not feasible. Each approach doesn't work in some cases, e.g. constructing messages in the `FunctionTypeSerializer` will not work for `lambda`s defined in the module:
 ```
Error during serializing test_one.models.<lambda>:  ...
```
or imported from other modules:
```
ValueError: Error during serializing test_one.utils.<lambda>: ...
```

I think we should close this as wontfix :disappointed: